frontend/src/components/NoteView/EditorMenuBar.tsx

```diff
@@ -0,0 +1,135 @@
+import { RxUnderline } from "react-icons/rx";
+import { GoCheckbox } from "react-icons/go";
+
+interface EditorMenuBarProps {
+  editor: Editor;
+}
+
+const HEADING_NUMBERS = [1, 2, 3, 4, 5, 6];
+const HIGHLIGHT_COLOR = [
+  { colorName: "Blue", colorCode: "#9fd0f6" },
+  { colorName: "Purple", colorCode: "#c9b9f9" },
+  { colorName: "Red", colorCode: "#fbc5c5" },
+  { colorName: "Yellow", colorCode: "#f9f4a2" },
+  { colorName: "Orange", colorCode: "#fbc88e" },
+];
+
+function EditorMenuBar({ editor }: EditorMenuBarProps) {
+  const [shouldShowHeadingBtns, setShouldShowHeadingBtns] = useState(false);
+  const [shouldShowHighlightColorBtns, setShouldShowHighlightColorBtns] =
+    useState(false);
+  const [shouldShowListBtns, setShouldShowListBtns] = useState(false);
+
+  const headingBtns = (HEADING_NUMBERS as Level[]).map((headingNum) => {
+    return (
+      <button
+        onClick={() => {
+          setShouldShowHeadingBtns(false);
+          editor.chain().focus().toggleHeading({ level: headingNum }).run();
+        }}
+        className={
+          editor.isActive("heading", { level: headingNum }) ? "active" : ""
+        }
+      >
+        Heading {headingNum}
+      </button>
+    );
+  });
+
+  const HighlightBtns = HIGHLIGHT_COLOR.map((colorObj) => {
+    const { colorName, colorCode } = colorObj;
+    return (
+      <button
+        onClick={() => {
+          setShouldShowHighlightColorBtns(false);
+          editor.chain().focus().toggleHighlight({ color: colorCode }).run();
+        }}
+        className={
+          editor.isActive("highlight", { color: colorCode }) ? "active" : ""
+        }
+      >
+        {colorName}
+      </button>
+    );
+  });
+
+  const ListBtns = (
+    <div>
+      <button
+        onClick={() => {
+          setShouldShowListBtns(false);
+          editor.chain().focus().toggleBulletList().run();
+        }}
+        className={editor.isActive("bulletList") ? "active" : ""}
+      >
+        List
+      </button>
+      <button
+        onClick={() => {
+          setShouldShowListBtns(false);
+          editor.chain().focus().toggleOrderedList().run();
+        }}
+        className={editor.isActive("orderedList") ? "active" : ""}
+      >
+        Ordered List
+      </button>
+    </div>
+  );
+
+  return (
+    <EditorMenuBarWrapper>
+      <div>{shouldShowHeadingBtns && headingBtns}</div>
+      <button onClick={() => setShouldShowHeadingBtns(true)}>
+        <FaHeading size={13} />
+        <MdExpandMore size={13} />
+      </button>
+      <button
+        onClick={() => editor.chain().focus().toggleTaskList().run()}
+        className={editor.isActive("taskList") ? "is-active" : ""}
+      >
+        <GoCheckbox size={18} />
+      </button>
+      <button
+        onClick={() => editor.chain().focus().toggleBold().run()}
+        disabled={!editor.can().chain().focus().toggleBold().run()}
+        className={editor.isActive("bold") ? "active" : ""}
```

Comment:
Please refactor these. Assign some constants/function for them.
```suggestion
        const focusedPosition = editor.chain().focus()
        const canFocusedPositionRun = editor.can().chain().focus()
        const getMarkClassName = (mark: string) => editor.isActive(mark) ? "active" : ""
         
        onClick={() => focusedPosition.toggleBold().run()}
        disabled={!canFocusedPositionRun.toggleBold().run()}
        className={getMarkClassName("bold")}
```